### DISCUSSION TITLE
How to clear stored permission settings?

### DISCUSSION THREAD

**Discussion by n303ef** [cefsharp/CefSharp#4198] [General]
With recently added PermissionHandler we are able to manage requested permissions (allow/deny). Problem is that these decisons are cached and OnShowPermissionPrompt/OnRequestMediaAccessPermission methods are not fired again until we dispose BrowserContext and create a new one. Is there any way to clear cached permission settings? Similarly as we can clear HTTP credentials cache by calling BrowserContext.ClearHttpAuthCredentials().

**Comment by amaitland** [comment]
> Similarly as we can clear HTTP credentials cache by calling BrowserContext.ClearHttpAuthCredentials().

CEF has yet to add a method for clearing permissions. CEF has it's own feature request forum at https://magpcss.org/ceforum/viewforum.php?f=7

Once CEF adds support then CefSharp can add the method.

In the meantime you can try using DevTools to clear permissions. 

- https://chromedevtools.github.io/devtools-protocol/tot/Browser/#method-resetPermissions
- http://cefsharp.github.io/api/103.0.x/html/M_CefSharp_DevTools_Browser_BrowserClient_ResetPermissionsAsync.htm

I've Not tested this so no idea if it works.

**Reply by n303ef** [reply]
I've tried DevTools first and unfortunately it doesn't help. Can be checked on this website: http://permission.site/
